test/utils.jl

```diff
@@ -1,33 +1,84 @@
 module TestUtils
 
 using Test, DataFrames
-#
-# make_unique(df::AbstractDataFrame, makeunique::Bool)
-#
+
 @testset "make_unique" begin
     @test DataFrames.make_unique([:x, :x, :x_1, :x2], makeunique=true) == [:x, :x_2, :x_1, :x2]
     @test_throws ArgumentError DataFrames.make_unique([:x, :x, :x_1, :x2], makeunique=false)
     @test DataFrames.make_unique([:x, :x_1, :x2], makeunique=false) == [:x, :x_1, :x2]
 end
-#
-# repeat(df::AbstractDataFrame, count::Integer)
-#
-@testset "count" begin
+
+@testset "repeat count" begin
     df = DataFrame(a = 1:2, b = 3:4)
     ref = DataFrame(a = repeat(1:2, 2),
                     b = repeat(3:4, 2))
     @test repeat(df, 2) == ref
     @test repeat(view(df, 1:2, :), 2) == ref
+
+    @test size(repeat(df, 0)) == (0, 2)
+    @test size(repeat(df, false)) == (0, 2)
+    @test_throws ArgumentError repeat(df, -1)
 end
-#
-# repeat(df::AbstractDataFrame; inner::Integer = 1, outer::Integer = 1)
-#
-@testset "inner_outer" begin
+
+@testset "repeat inner_outer" begin
     df = DataFrame(a = 1:2, b = 3:4)
     ref = DataFrame(a = repeat(1:2, inner = 2, outer = 3),
                     b = repeat(3:4, inner = 2, outer = 3))
     @test repeat(df, inner = 2, outer = 3) == ref
     @test repeat(view(df, 1:2, :), inner = 2, outer = 3) == ref
+
+    @test size(repeat(df, inner = 2, outer = 0)) == (0, 2)
+    @test size(repeat(df, inner = 0, outer = 3)) == (0, 2)
+    @test size(repeat(df, inner = 2, outer = false)) == (0, 2)
+    @test size(repeat(df, inner = false, outer = 3)) == (0, 2)
+
+    @test_throws ArgumentError repeat(df, inner = 2, outer = -1)
+    @test_throws ArgumentError repeat(df, inner = -1, outer = 3)
+end
+
+@testset "repeat! count" begin
+    df = DataFrame(a = 1:2, b = 3:4)
+    ref = DataFrame(a = repeat(1:2, 2),
+                    b = repeat(3:4, 2))
+    repeat!(df, 2)
+    @test df == ref
+
+    for v in (0, false)
+        df = DataFrame(a = 1:2, b = 3:4)
+        repeat!(df, v)
+        @test size(df) == (0, 2)
+    end
+
+    df = DataFrame(a = 1:2, b = 3:4)
+    @test_throws ArgumentError repeat(df, -1)
+    @test df == DataFrame(a = 1:2, b = 3:4)
+
+    @test_throws MethodError repeat!(view(df, 1:2, :), 2)
+end
+
+@testset "repeat! inner_outer" begin
+    df = DataFrame(a = 1:2, b = 3:4)
+    ref = DataFrame(a = repeat(1:2, inner = 2, outer = 3),
+                    b = repeat(3:4, inner = 2, outer = 3))
+    repeat!(df, inner = 2, outer = 3)
+    @test df == ref
```

Comment:
Also test that column vectors are `===` or `!==` from original column vectors (depending on our decision)?

Reply:
OK - added.